ouroboros-network/test/Test/Mux.hs

```diff
@@ -46,14 +53,58 @@ tests =
   , testProperty "too short header"     prop_mux_short_header
   ]
 
-version0 :: Mx.Version
-version0 = Mx.Version0  (Mx.NetworkMagic 0xcafebeeb)
-
-version0' :: Mx.Version
-version0' = Mx.Version0  (Mx.NetworkMagic 0xa5a5a5a5)
-
-version1 :: Mx.Version
-version1 = Mx.Version1  (Mx.NetworkMagic 0xbeebcafe)
+newtype NetworkMagic = NetworkMagic Word32
+  deriving (Eq, Ord, Show)
+
+instance Arbitrary NetworkMagic where
+    arbitrary = NetworkMagic <$> arbitrary
+    shrink (NetworkMagic nm) = map NetworkMagic (shrink nm)
+
+instance Serialise NetworkMagic where
+  encode (NetworkMagic nm) = encode nm
+  decode = NetworkMagic <$> CBOR.decodeWord32
+
+type instance Mx.MuxVersion 0 = NetworkMagic
+
+data NetworkMagicWithBufSize = NetworkMagicWithBufSize Word32 Word16
+  deriving (Eq, Ord, Show)
+
```

Comment:
I don't know if this is a toy example only, but the buffer size should be larger than the maximum blocksize so 16bits isn't enough.

Reply:
This is a toy example, just to have two different types. But maybe that's not the best choice since they will be both represented by `Int`s as chief terms.